external/xmlrpc/xmlparser.c

```diff
@@ -140,6 +140,10 @@ static int xmlrpc_getelement(struct parsebuf_s *pbuf, char *data, int dataSize)
 		return XMLRPC_PARSE_ERROR;
 	}
 
+	if (pbuf->len < 0 || pbuf->index < 0) {
```

Comment:
Do we really need this check ? As i observe pbuf->index  is initialized to zero in xmlrpc_parse() API and it's been always incremented and thus it can at least never become negative value. 
Since pbuf->index can not become negative , even though pbuf->len becomes negative , that case has been already handled in above if check at line no 139  ( -1 <= 0 )
Kindly check again.

Or did you encounter this error in any specific test case?

Reply:
I think you might be right. But I am not sure that pbuf->index is set 0 at all cases. I will decide to close it or not after checking

Reply:
This is external open source. Please keep modification as minimum as possible.